dbaccess/transaction.go

```diff
@@ -98,40 +92,41 @@ func TransactionsByAddressCount(ctx Context, address string) (uint64, error) {
 		return 0, err
 	}
 
-	var count uint64
 	query := db.Model(&dbmodels.Transaction{})
-	dbResult := joinTxInputsTxOutputsAndAddresses(query).
-		Where("`out_addresses`.`address` = ?", address).
-		Or("`in_addresses`.`address` = ?", address).
-		Count(&count)
-
-	dbErrors := dbResult.GetErrors()
-	if httpserverutils.HasDBError(dbErrors) {
-		return 0, httpserverutils.NewErrorFromDBErrors("some errors were encountered when counting transactions in the database:", dbErrors)
+	count, err := joinTxInputsTxOutputsAndAddresses(query).
+		Where("out_addresses.address = ?", address).
+		WhereOr("in_addresses.address = ?", address).
+		Count()
+
+	if err != nil {
+		return 0, err
 	}
 
-	return count, nil
+	return uint64(count), nil
 }
 
 // AcceptedTransactionsByBlockHashes retrieves a list of transactions that were accepted
 // by blocks with the given `blockHashes`
 func AcceptedTransactionsByBlockHashes(ctx Context, blockHashes []string, preloadedFields ...dbmodels.FieldName) ([]*dbmodels.Transaction, error) {
+	if len(blockHashes) == 0 {
+		return nil, nil
+	}
+
 	db, err := ctx.db()
 	if err != nil {
 		return nil, err
 	}
+	var transactions []*dbmodels.Transaction
+	query := db.Model(&transactions).
+		ColumnExpr("DISTINCT(transaction.id)").
```

Comment:
Why sometimes you do `err = query.Select("DISTINCT transactions.*")` and sometimes you do `ColumnExpr("DISTINCT(transaction.id)")`? Can you explain the difference?

Reply:
This was a mistake `ColumnExpr` is the right way to go